_posts/2020-10-19-torch-image-classification/torch-image-classification.Rmd

```diff
@@ -0,0 +1,543 @@
+---
+title: "Classifying images with torch"
+description: > 
+ We learn about transfer learning, input pipelines, and learning rate schedulers, all while using torch to tell apart species of beautiful birds.
+author:
+  - name: Sigrid Keydana
+    affiliation: RStudio
+    affiliation_url: https://www.rstudio.com/
+slug: keydanatorchimageclassification
+date: 10-19-2020
+categories:
+  - Torch
+  - R
+  - Image Recognition & Image Processing
+bibliography: bibliography.bib
+output:
+  distill::distill_article:
+    self_contained: false
+    toc: true
+preview: images/image_classif_birds.png
+---
+
+```{r setup, include=FALSE}
+knitr::opts_chunk$set(echo = TRUE, eval = FALSE)
+```
+
+In recent posts, we've been exploring essential `torch` functionality: [tensors](https://blogs.rstudio.com/ai/posts/2020-10-01-torch-network-from-scratch/), the sine qua non of every deep learning framework; [autograd](https://blogs.rstudio.com/ai/posts/2020-10-05-torch-network-with-autograd), `torch`'s implementation of reverse-mode automatic differentiation; [modules](https://blogs.rstudio.com/ai/posts/2020-10-07-torch-modules), composable building blocks of neural networks; and [optimizers](https://blogs.rstudio.com/ai/posts/2020-10-09-torch-optim/), the -- well -- optimization algorithms that `torch` provides.
+
+But we haven't really had our "hello world" moment yet, at least not if by "hello world" you mean the inevitable *deep learning experience of classifying pets*. Cat or dog? Beagle or boxer? Chinook or chihuahua? We'll distinguish ourselves by asking a (slightly) different question: What kind of bird?
+
+Questions we'll address on our way:
+
+-   What are the core roles of `torch` *datasets* and *data loaders*, respectively?
+
+-   How to apply `transform`s, both for image preprocessing and data augmentation.
+
+-   How to use Resnet [@HeZRS15], a pre-trained model that comes with `torchvision`, for transfer learning.
+
+-   How to use learning rate schedulers, and in particular, the one-cycle learning rate algorithm [\@abs-1708-07120].
+
+-   How to find a good initial learning rate.
+
+For convenience, the code is available on [Google Colaboratory](https://colab.research.google.com/drive/1OJzzqiQVbh3ZdLB2L2t_DhBGInlh9o-k?usp=sharing) -- no copy-pasting required.
+
+## Data loading and preprocessing
+
+The example dataset used here is available on [Kaggle](https://www.kaggle.com/gpiosenka/100-bird-species/data){.uri}.
+
+Conveniently, it may be obtained using [torchdatasets](https://github.com/mlverse/torchdatasets), which uses [`pins`](https://github.com/rstudio/pins) for authentication, retrieval and storage. To enable `pins` to manage your Kaggle downloads, please follow the instructions [here](https://pins.rstudio.com/articles/boards-kaggle.html).
```

Comment:
```suggestion
Conveniently, it may be obtained using [`torchdatasets`](https://github.com/mlverse/torchdatasets), which uses [`pins`](https://github.com/rstudio/pins) for authentication, retrieval and storage. To enable `pins` to manage your Kaggle downloads, please follow the instructions [here](https://pins.rstudio.com/articles/boards-kaggle.html).
```